### PR TITLE
Add StringReader provider

### PR DESCRIPTION
**Pull request by mexx**
relates #71 
without encoding support

### CODE REVIEW

**Comment by mexx** [comment]
Ping 🏓 @vasily-kirichenko @sergey-tihon @forki

**Comment by vasily-kirichenko** [comment]
What about the review?

**Comment by dsyme** [comment]
> What about the review?

@vasily-kirichenko Which review? thx

**Comment by vasily-kirichenko** [comment]
This one:

image

Can you see it, guys cannot you?..

**Comment by sergey-tihon** [comment]
@vasily-kirichenko Seems like you did not complete (publish) review. We see nothing

**Comment by vasily-kirichenko** [comment]
Damn. Github is too awesome for me :(

File: src/FSharp.Management/StringReaderProvider.fs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        parameters =
+            [ ProvidedStaticParameter("path", typeof<string>) ],
+        instantiationFunction = (fun typeName parameterValues ->
+            match parameterValues with
+            | [| :? string as path |] ->
+                let filePath =
+                    match Path.IsPathRooted(path) with
+                    | false -> Path.Combine(resolutionFolder, path)
+                    | true -> path
+                match File.Exists(filePath) with
```
Comment (mexx) [inline src/FSharp.Management/StringReaderProvider.fs]: Same control structure as in `FileSystemProviver`.
For me a pattern match reads better, but sure it's a matter of taste.
Which style should I follow?

File: src/FSharp.Management/StringReaderProvider.fs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                | false -> failwith (sprintf "Specified file [%s] could not be found" path)
```
Comment (mexx) [inline src/FSharp.Management/StringReaderProvider.fs]: Copied over from `FileSystemProvider`. Will change, as not error prone any more with the warning for extra arguments on `falwithf`.

File: tests/FSharp.Management.Tests/StringReaderProvider.Tests.data [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+Etiam pulvinar dui a eros ullamcorper luctus. Nam lobortis urna vitae lorem maximus, ac rhoncus felis ultricies. Morbi vestibulum ultricies urna, sed suscipit velit vehicula in. Sed iaculis velit nec ligula cursus semper. Nam sed felis vitae dui aliquet interdum sed sed eros. Donec porta ex a dolor blandit, non tempor purus molestie. Pellentesque viverra diam quis fringilla commodo. Praesent ut enim ac massa imperdiet pellentesque in eget nunc. Curabitur imperdiet erat sit amet condimentum maximus. Proin tincidunt sem sed lorem congue, nec scelerisque libero viverra.
```
Comment (mexx) [inline tests/FSharp.Management.Tests/StringReaderProvider.Tests.data]: What's wrong with Latin, especially with all beloved `Lorem ipsum`?

Comment (vasily-kirichenko) [inline tests/FSharp.Management.Tests/StringReaderProvider.Tests.data]: What about Chinese?

Comment (mexx) [inline tests/FSharp.Management.Tests/StringReaderProvider.Tests.data]: I don't get the point, it's just test data. 👎

Comment (vasily-kirichenko) [inline tests/FSharp.Management.Tests/StringReaderProvider.Tests.data]: After a half of a year someone will read this test and will be asking himself "Why was Latin used here? Does it make the TP work differently compared to Chinese?"

Comment (mexx) [inline tests/FSharp.Management.Tests/StringReaderProvider.Tests.data]: Sorry, but same argument applies to English as well.
Why not just use `русский`? давай, а?

Comment (vasily-kirichenko) [inline tests/FSharp.Management.Tests/StringReaderProvider.Tests.data]: Just replace it with English. We should use other languages for a purpose, I don't see one here.

**Comment by vasily-kirichenko** [comment]
Thanks for contribution!

**Pull request merged**